contracts/v2/marketplace/TokenMarketplaceV2.sol

```diff
@@ -0,0 +1,509 @@
+    _;
+  }
+
+  modifier onlyWhenBidOverMinAmount(uint256 _tokenId) {
+    require(msg.value >= offers[_tokenId].offer.add(minBidAmount), "Offer not enough");
+    _;
+  }
+
+  modifier onlyWhenTokenAuctionEnabled(uint256 _tokenId) {
+    require(!disabledTokens[_tokenId], "Token not enabled for offers");
+    _;
+  }
+
+  modifier nonReentrant() {
+    // On the first call to nonReentrant, _notEntered will be true
+    require(_notEntered, "ReentrancyGuard: reentrant call");
+
+    // Any calls to nonReentrant after this point will fail
+    _notEntered = false;
+
+    _;
+
+    // By storing the original value once again, a refund is triggered (see
+    // https://eips.ethereum.org/EIPS/eip-2200)
+    _notEntered = true;
+  }
+
+  /////////////////
+  // Constructor //
+  /////////////////
+
+  // Set the caller as the default KO account
+  constructor(IKODAV2Methods _kodaAddress, address _koCommissionAccount) public {
+    kodaAddress = _kodaAddress;
+    koCommissionAccount = _koCommissionAccount;
+    super.addAddressToWhitelist(msg.sender);
+  }
+
+  //////////////////////////
+  // User Bidding Actions //
+  //////////////////////////
+
+  function placeBid(uint256 _tokenId)
+  public
+  payable
+  whenNotPaused
+  nonReentrant
+  onlyWhenTokenExists(_tokenId)
+  onlyWhenBidOverMinAmount(_tokenId)
+  onlyWhenTokenAuctionEnabled(_tokenId)
+  {
+    require(!isContract(msg.sender), "Unable to place a bid as a contract");
+    _refundHighestBidder(_tokenId);
+
+    offers[_tokenId] = Offer(msg.sender, msg.value);
+
+    address currentOwner = kodaAddress.ownerOf(_tokenId);
+
+    emit BidPlaced(_tokenId, currentOwner, msg.sender, msg.value);
+  }
+
+  function withdrawBid(uint256 _tokenId)
+  public
+  whenNotPaused
+  nonReentrant
+  onlyWhenTokenExists(_tokenId)
+  onlyWhenOfferOwner(_tokenId)
+  {
+    _refundHighestBidder(_tokenId);
+
+    emit BidWithdrawn(_tokenId, msg.sender);
+  }
+
+  function rejectBid(uint256 _tokenId)
+  public
+  whenNotPaused
+  nonReentrant
+  {
+    address currentOwner = kodaAddress.ownerOf(_tokenId);
+    require(currentOwner == msg.sender, "Not token owner");
+
+    uint256 currentHighestBiddersAmount = offers[_tokenId].offer;
+    require(currentHighestBiddersAmount > 0, "No offer open");
+
+    address currentHighestBidder = offers[_tokenId].bidder;
+
+    _refundHighestBidder(_tokenId);
+
+    emit BidRejected(_tokenId, currentOwner, currentHighestBidder, currentHighestBiddersAmount);
+  }
+
+  function acceptBid(uint256 _tokenId, uint256 _acceptedAmount)
+  public
+  whenNotPaused
+  nonReentrant
+  {
+    address currentOwner = kodaAddress.ownerOf(_tokenId);
+    require(currentOwner == msg.sender, "Not token owner");
+
+    Offer storage offer = offers[_tokenId];
+
+    uint256 winningOffer = offer.offer;
+    // Check valid offer and offer not replaced whilst inflight
+    require(winningOffer > 0 && winningOffer >= _acceptedAmount, "Offer amount not satisfied");
```

Comment:
should it not be `_acceptedAmount >= winningOffer`? otherwise your offer could be 500, you see artist accepts for 200 and then front-run the accept tx to drop your offer to 200